src/edfi-paging-test/edfi_paging_test/api/paginated_result.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from typing import Any, Dict, List, Optional
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from edfi_paging_test.api.request_client import RequestClient
+
+
+class PaginatedResult():
+    """
+    The PaginatedResult class is bound with information from the response of the
+    api, specifically when it returns a list of items that could be paginated.
+    Parameters
+    ----------
+    request_client : RequestClient
+        The request client.
+    page_size : int
+        The number of items per page.
+    total_count : int
+        Total number of items.
+    api_response : dict
+        The original response as a dictionary.
+    resource_name : str
+        The name used by the API for the current resource.
+    current_page : Optional[str]
+        The page that you have requested.
+    Attributes
+    ----------
+    current_page_items : Dict[str, Any]
+        The list of items for the current page.
+    """
+
+    def __init__(
+        self,
+        request_client: "RequestClient",
```

Comment:
```suggestion
        request_client: RequestClient,
```

That ought to work...

Reply:
can't not import RequestClient since it causes circular import

Reply:
Hm... wonder if there's a way we can restructure the code to avoid a circular import?

Can you add a comment explaining why the unusual line?